src/services/assetService.ts

```diff
@@ -149,8 +149,12 @@ export class AssetService {
     public async save(metadata: IAssetMetadata): Promise<IAssetMetadata> {
         Guard.null(metadata);
 
-        const fileName = `${metadata.asset.id}${constants.assetMetadataFileExtension}`;
-        await this.storageProvider.writeText(fileName, JSON.stringify(metadata, null, 4));
+        // Only save asset metadata if asset is in a tagged state
```

Comment:
Perf improvement: No need to store an asset metadata JSON file if the asset has not been tagged - otherwise the core asset info is already going to be inside the project file.